src/apps/chatter/components/chatbox/chatbox.js

```diff
@@ -18,12 +30,38 @@ export const ChatBox = ({ roomId }) => {
     window.scrollTo(0, document.body.scrollHeight)
   }, [entries])
 
-  return (
-    <div className='chatbox'>
-      {entries.map(Entry)}
-      <EntryInput
-        onSend={handleSend}
-      />
-    </div>
-  )
+  const enterPasswordJsx = <div className='chatbox'>
+    <TextInput
+      name='password'
+      placeholder='Enter password...'
+      value={passwordInput}
+      onChange={handleInputChange}
+    />
+    <button
+      className='btn'
+      onClick={handleSubmitPassword}
+      disabled={passwordInput === ''}
+    >Submit</button>
+  </div>
+
+  const chatJsx = <div className='chatbox'>
+    {entries.map(Entry)}
+    <EntryInput
+      onSend={handleSend}
+    />
+  </div>
+
+  const loadingJsx = <div className='chatbox'>
+    <LoadingIndicator />
+  </div>
```

Comment:
You should define child components as outside of this component. Currently, these are redefined on every render of ChatBox, even though only 1 will be rendered at a time.

You are also repeating the `<div className='chatbox'>` 3 times. Seems it could just always be rendered by ChatBox.

Reply:
Yes.